metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -91,6 +91,31 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
             else:
                 self.report.report_dropped(t)
 
+    def get_schema_str_replace_confluent_ref_avro(
+        self, schema: Schema, schema_seen: Dict = None
+    ) -> str:
+        if not schema.references:
+            return schema.schema_str
+
+        if schema_seen is None:
+            schema_seen = dict()
+        schema_str = schema.schema_str
+        for schema_ref in schema.references:
+            ref_subject = schema_ref["subject"]
+            ref_name = schema_ref["name"]
+            if ref_subject in schema_seen:
+                continue
+            tmp_schema = self.schema_registry_client.get_latest_version(ref_subject)
+            schema_seen[ref_subject] = tmp_schema
+            logger.debug(f"ref for {ref_subject} is {tmp_schema.schema.schema_str}")
+            schema_str = schema_str.replace(
+                f'"{ref_name}"',
```

Comment:
Do the reference names always appear in double quotes? Can they be single quoted? What if this string appears in documentation? Does `confluent_kafka` provider some a parser for us to more deterministically replace the reference schemas names with their schema definitions to generate a well-formed AVRO schema?

Reply:
Avro has to be valid JSON so this should be in double quotes always. `confluent_kafka` does not provide any such method.